nobl9/resource_agent.go

```diff
@@ -468,6 +471,47 @@ func marshalAgentAppDynamics(d *schema.ResourceData, diags diag.Diagnostics) *v1
 	}
 }
 
+/**
+ * Azure Monitor Agent
+ * https://docs.nobl9.com/Sources/azure-monitor#azure-monitor-agent
+ */
+const azureMonitorAgentType = "azuremonitor"
+const azureMonitorAgentConfigKey = "azuremonitor_config"
+
+func schemaAgentAzureMonitor() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		azureMonitorAgentConfigKey: {
+			Type:        schema.TypeSet,
+			Optional:    true,
+			Description: "[Configuration documentation](https://docs.nobl9.com/Sources/azure-monitor#azure-monitor-agent)",
+			MinItems:    1,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"tenant_id": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Azure Tenant Id.",
+					},
+				},
+			},
+		},
+	}
+}
+
+func marshalAgentAzureMonitor(d *schema.ResourceData, diags diag.Diagnostics) *v1alpha.AzureMonitorAgentConfig {
+	data := getAgentResourceData(d, azureMonitorAgentType, azureMonitorAgentConfigKey, diags)
+
+	if data == nil {
+		return nil
+	}
+
+	tenantID := data["tenant_id"].(string)
```

Comment:
~~We could check if it was cast correctly.~~ I see we never do this in other integrations, so maybe it's fine.